File: 프로그래머스/Lv.1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    dic = {}
    count = 0
    for num in nums:
        if num not in dic:
            dic[num] = 1
            count += 1
        else:
            dic[num] = dic[num] + 1
    if count < len(nums) // 2:
        return count
    else:
        return len(nums) // 2

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        int cnt = 0;
        for (int key : map.keySet()) {
            cnt += 1;
        }
        int n = nums.length;
        
        if (n / 2 >= cnt) {
            answer = cnt;
        } else {
            answer = n / 2;
        }
        
        return answer;
    }
}